GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1795,26 +1795,48 @@ private void CurrentBranchDropDownOpening(object sender, EventArgs e)
         {
             branchSelect.DropDownItems.Clear();
 
-            ToolStripMenuItem item = new ToolStripMenuItem(checkoutBranchToolStripMenuItem.Text);
-            item.ShortcutKeys = checkoutBranchToolStripMenuItem.ShortcutKeys;
-            item.ShortcutKeyDisplayString = checkoutBranchToolStripMenuItem.ShortcutKeyDisplayString;
-            branchSelect.DropDownItems.Add(item);
-            item.Click += (hs, he) => CheckoutBranchToolStripMenuItemClick(hs, he);
-
+            AddCheckoutBranchMenuItem();
             branchSelect.DropDownItems.Add(new ToolStripSeparator());
+            AddBranchesMenuItems();
+
+            PreventToolStripSplitButtonClosing(sender as ToolStripSplitButton);
+        }
+
+        private void AddCheckoutBranchMenuItem()
+        {
+            var checkoutBranchItem = new ToolStripMenuItem(checkoutBranchToolStripMenuItem.Text)
+            {
+                ShortcutKeys = checkoutBranchToolStripMenuItem.ShortcutKeys,
+                ShortcutKeyDisplayString = checkoutBranchToolStripMenuItem.ShortcutKeyDisplayString
+            };
+            branchSelect.DropDownItems.Add(checkoutBranchItem);
+            checkoutBranchItem.Click += CheckoutBranchToolStripMenuItemClick;
+        }
 
-            foreach (var branch in Module.GetRefs(false))
+        private void AddBranchesMenuItems()
+        {
+            foreach (var branchName in GetBranchNames())
             {
-                var toolStripItem = branchSelect.DropDownItems.Add(branch.Name);
+                var toolStripItem = branchSelect.DropDownItems.Add(branchName);
                 toolStripItem.Click += BranchSelectToolStripItem_Click;
 
-                //Make sure there are never more than 100 branches added to the menu
-                //GitExtensions will hang when the drop down is to large...
-                if (branchSelect.DropDownItems.Count > 100)
-                    break;
             }
+        }
 
-            PreventToolStripSplitButtonClosing(sender as ToolStripSplitButton);
+        private IEnumerable<string> GetBranchNames()
+        {
+            var branches = Module.GetRefs(false);
+            var branchNames = branches.Select(b => b.Name);
+            if (Settings.BranchOrderingCriteria == BranchOrdering.Alphabetically)
+            {
+                branchNames = branchNames.OrderBy(b => b);
+            }
+
+            //Make sure there are never more than 100 branches added to the menu
+            //GitExtensions will hang when the drop down is to large...
```

Comment:
grammar:
Make sure there are never more than 100 branches added to the menu
->
Make sure there are never more than a 100 branches added to the menu

GitExtensions will hang when the drop down is to large
->
GitExtensions will hang when the drop down is too large

Reply:
Fixed